source/s3_meta_request.c

```diff
@@ -83,6 +83,10 @@ struct aws_s3_client *aws_s3_meta_request_acquire_client(struct aws_s3_meta_requ
     client = meta_request->synced_data.client;
 
     if (client != NULL) {
+        AWS_LOGF_TRACE(
+            AWS_LS_S3_META_REQUEST,
+            "id=%p Meta request trying to get reference to client, but client is null.",
```

Comment:
The reference to the client goes away when the meta request finishes.  Because of that, it's not invalid to get a NULL client ref back.

But, all places that currently need this, and all places that we should be using it in the future, we could expect that the client is not NULL.  So we could probably make this, and the other existing place that allows a NULL right now, an assert.